client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -228,9 +232,9 @@ const Checkout = React.createClass( {
 
 			return purchasePaths.managePurchase( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId );
 		} else if ( cartItems.hasFreeTrial( this.props.cart ) ) {
-			this.props.clearSitePlans( this.props.sites.getSelectedSite().ID );
+			this.props.clearSitePlans( selectedSite.ID );
 
-			return `/plans/${ this.props.sites.getSelectedSite().slug }/thank-you`;
+			return `/plans/${ selectedSite.slug }/thank-you`;
```

Comment:
We didn't have the proper bulletproofing before, but we should at least avoid js errors when accessing properties from things that haven't been loaded yet. 

We can change which route we redirect to as well, but I'd save that for another PR. cc @drewblaisdell